infrastructure/create-github-environment.js

```diff
@@ -0,0 +1,226 @@
+    INFOBIP_API_KEY: process.env.INFOBIP_API_KEY || '',
+    INFOBIP_GATEWAY_ENDPOINT: process.env.INFOBIP_GATEWAY_ENDPOINT || '',
+    INFOBIP_SENDER_ID: process.env.INFOBIP_SENDER_ID || '' // the name of the SMS sender e.g. OpenCRVS
+  },
+  seeding: {
+    ACTIVATE_USERS: true,
+    AUTH_HOST: '',
+    COUNTRY_CONFIG_HOST: '',
+    GATEWAY_HOST: ''
+  },
+  smtp: {
+    SMTP_HOST: process.env.SMTP_HOST || '',
+    SMTP_USERNAME: process.env.SMTP_USERNAME || '',
+    SMTP_PASSWORD: process.env.SMTP_PASSWORD || '',
+    EMAIL_API_KEY: process.env.EMAIL_API_KEY || '',
+    SMTP_PORT: '',
+    ALERT_EMAIL: ''
+  },
+  vpn: {
+    // openconnect details for optional VPN
+    VPN_PROTOCOL: '', // e,g, fortinet, wireguard etc
+    VPN_HOST: process.env.VPN_HOST || '',
+    VPN_PORT: process.env.VPN_PORT || '',
+    VPN_USER: process.env.VPN_USER || '',
+    VPN_PWD: process.env.VPN_PWD || '',
+    VPN_SERVERCERT: process.env.VPN_SERVERCERT || ''
+  },
+  whitelist: {
+    CONTENT_SECURITY_POLICY_WILDCARD: '', // e.g. *.<your-domain>
+    CLIENT_APP_URL: '',
+    LOGIN_URL: ''
+  },
+  backup: {
+    BACKUP_HOST: process.env.BACKUP_HOST || '',
+    BACKUP_DIRECTORY: '',
+    qa: {
+      RESTORE_DIRECTORY: '' // If making use of script to restore a production backup on QA for regular monitoring
+    }
+  }
+}
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN
+})
+
+async function createVariable(environment, name, value) {
+  await octokit.request(
+    `POST /repositories/${config.repo.REPOSITORY_ID}/environments/${config.environment}/variables`,
+    {
+      repository_id: config.repo.REPOSITORY_ID,
+      environment_name: environment,
+      name: name,
+      value: value,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    }
+  )
+}
+
+async function createSecret(environment, key, keyId, name, secret) {
+  //Check if libsodium is ready and then proceed.
+  await sodium.ready
+
+  // Convert Secret & Base64 key to Uint8Array.
+  let binkey = sodium.from_base64(key, sodium.base64_variants.ORIGINAL)
+  let binsec = sodium.from_string(secret)
+
+  //Encrypt the secret using LibSodium
+  let encBytes = sodium.crypto_box_seal(binsec, binkey)
+
+  // Convert encrypted Uint8Array to Base64
+  const encryptedValue = sodium.to_base64(
+    encBytes,
+    sodium.base64_variants.ORIGINAL
+  )
+
+  await octokit.request(
+    `PUT /repositories/${config.repo.REPOSITORY_ID}/environments/${environment}/secrets/${name}`,
+    {
+      repository_id: config.repo.REPOSITORY_ID,
+      environment_name: environment,
+      secret_name: name,
+      encrypted_value: encryptedValue,
+      key_id: keyId,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    }
+  )
+}
+
+async function getPublicKey(environment) {
+  await octokit.request(
+    `PUT /repos/${config.repo.DOCKERHUB_ACCOUNT}/${config.repo.DOCKERHUB_REPO}/environments/${environment}`,
```

Comment:
(node:44598) UnhandledPromiseRejectionWarning: HttpError: Not Found

should this be REPOSITORY_ID  instead of dockerhub accounts?